metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema.py

```diff
@@ -86,6 +106,64 @@ class SnowflakeDatabase:
     comment: Optional[str]
     last_altered: Optional[datetime] = None
     schemas: List[SnowflakeSchema] = field(default_factory=list)
+    tags: Optional[List[SnowflakeTag]] = None
+
+
+class _SnowflakeTagCache:
+    def __init__(self) -> None:
+        self._database_tags: Dict[str, List[SnowflakeTag]] = defaultdict(list)
+        self._schema_tags: Dict[str, Dict[str, List[SnowflakeTag]]] = defaultdict(
+            lambda: defaultdict(list)
+        )
+        self._table_tags: Dict[
```

Comment:
some quick comments about structure of these variables would be helpful

e.g. `self._column_tags[database][schema][table_or_view][column] = <list of tags applied to this column>`